scripts/fetchAndSubmitAddMemberEvents.js

```diff
@@ -0,0 +1,147 @@
+import dotenv from 'dotenv'
+import ora from 'ora'
+import * as ethers from 'ethers'
+import * as fs from 'fs/promises'
+import yargs from 'yargs/yargs'
+import axios from 'axios'
+
+dotenv.config()
+
+const argv = yargs(process.argv.slice(2))
+    .env(true)
+    .option('r', {
+        alias: 'rpc-url',
+        default: 'http://localhost:8545',
+        desc: 'The RPC URL to connect to (supports RPC_URL env var)'
+    })
+    .option('s', {
+        alias: 'sequencer-url',
+        desc: 'The sequencer url to use'
+    })
+    .option('a', {
+        alias: 'address',
+        default: '0xD81dE4BCEf43840a2883e5730d014630eA6b7c4A',
+        desc: 'The address of the smart contract to read events from'
+    })
+    .option('f', {
+        alias: 'from-block',
+        default: 39186981,
+        desc: 'The block to start reading events from'
+    })
+    .option('t', {
+        alias: 'to-block',
+        desc: 'If not set, will fetch the current block'
+    })
+    .option('b', {
+        alias: 'block-span',
+        default: 10_000,
+        desc: 'The number of blocks to read events from at a time'
+    })
+    .option('o', {
+        alias: 'output-file',
+        desc: 'If set, will write the events to a file'
+    })
+    .option('e', {
+        alias: 'error-output-file',
+        desc: 'If set, will errors that occurred during submission to a file'
+    })
+    .argv
+
+const semaphoreContract = new ethers.utils.Interface(
+    [
+        // This is the MemberAdded from ISemaphoreGroups.sol
+        // it seems to be the one we're actually emitting on the deployed smart contract
+        "event MemberAdded(uint256 indexed groupId, uint256 identityCommitment, uint256 root)",
+    ],
+)
+
+async function sendCommitmentsToSequencer(events) {
+    let commitments = events.map(e => {
+        const parsed = semaphoreContract.parseLog(e)
+        const identityCommitment = parsed.args[1]
+
+        return identityCommitment.toHexString()
+    })
+
+    return Promise.all(commitments.map(async (commitment) => {
+        try {
+            if (argv.sequencerUrl !== undefined) {
+                await axios.post(`${argv.sequencerUrl}/insertIdentity`, { identityCommitment: commitment })
+            }
```

Comment:
Is `sequencerUrl === undefined` an error condition? If so it should probably bail here, and if it _isn't_ it should at least warn.

Reply:
Not an error condition, I also envision a case where we don't provide sequencerUrl to first get the commitments into a file. But given that I should probably add an option to also read form a file...